src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -152,13 +173,92 @@ public X509Certificate[] getAcceptedIssuers() {
+            @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_TLS_KEYSTORE_TYPE) String keyStoreType,
+            @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_TLS_CERT_ALIAS) String certAlias,
+            @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_TLS_CERT_FILE)
+                    Optional<String> certFilePath,
+            @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_TLS_CERT_TYPE) String certType) {
+        boolean ssl =
+                keyStorePassFile.isPresent()
+                        && keyStoreFilePath.isPresent()
+                        && certFilePath.isPresent();
+        if (!ssl) {
+            if (keyStorePassFile.isPresent()
+                    || keyStoreFilePath.isPresent()
+                    || certFilePath.isPresent()) {
+                throw new IllegalArgumentException(
+                        "The file paths for the keystore, keystore password, and certificate must"
+                            + " ALL be provided to set up HTTPS connections. Otherwise, make sure"
+                            + " they are all unset to use an HTTP server.");
+            }
+            return Optional.empty();
+        }
+
+        try (InputStream pass = new FileInputStream(keyStorePassFile.get());
+                InputStream keystore = new FileInputStream(keyStoreFilePath.get());
+                InputStream certFile = new FileInputStream(certFilePath.get())) {
+            SSLContext sslContext = SSLContext.getInstance(serverTlsVersion);
+
+            // initialize keystore
+            String password = IOUtils.toString(pass, Charset.forName(passFileCharset));
+            password = password.substring(0, password.length() - 1);
+            KeyStore ks = KeyStore.getInstance(keyStoreType);
+            ks.load(keystore, password.toCharArray());
+
+            // set up certificate factory
+            CertificateFactory cf = CertificateFactory.getInstance(certType);
+            Certificate cert = cf.generateCertificate(certFile);
+            if (ks.containsAlias(certAlias)) {
+                throw new IllegalStateException(
+                        String.format(
+                                "%s keystore at %s already contains a certificate with alias"
+                                        + " \"%s\"",
+                                keyStoreType, keyStoreFilePath, certAlias));
+            }
+            ks.setCertificateEntry(certAlias, cert);
+
+            // set up key manager factory
+            KeyManagerFactory kmf =
+                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(ks, password.toCharArray());
+
+            // set up trust manager factory
+            TrustManagerFactory tmf =
+                    TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tmf.init(ks);
+
+            // set up HTTPS context
+            sslContext.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
```

Comment:
On line 169 when we set up the SSLContext for the Agent's HTTP Client, we supply a `new SecureRandom()` here. I think this ends up being the same thing in effect, since it looks like the JDK library code will also just call `new SecureRandom()` if `null` is passed, so maybe that other line needs to be adjusted to match this one.